extensions/vertx/runtime/src/main/java/io/quarkus/vertx/runtime/VertxProducer.java

```diff
@@ -1,35 +1,33 @@
 package io.quarkus.vertx.runtime;
 
+import javax.annotation.PostConstruct;
 import javax.enterprise.context.ApplicationScoped;
 import javax.enterprise.inject.Produces;
+import javax.inject.Inject;
 import javax.inject.Singleton;
 
 import io.vertx.core.Vertx;
-import io.vertx.core.eventbus.EventBus;
 
 /**
- * Produces a configured Vert.x instance.
- * It also exposes the Vert.x event bus.
+ * Produces Axle and Rx Vert.x instances.
+ * <p>
+ * The original Vert.x instance is coming from the common artifact.
  */
 @ApplicationScoped
 public class VertxProducer {
 
-    private volatile Vertx vertx;
-    private volatile io.vertx.axle.core.Vertx axleVertx;
-    private volatile io.vertx.reactivex.core.Vertx rxVertx;
+    @Inject
```

Comment:
The event bus producers should be moved there too.

Reply:
Done.